connections/README.md

```diff
@@ -21,122 +22,164 @@ and spec status.
 
 ## Table of Contents
 
-- [Connection Establishment in libp2p](#connection-establishment-in-libp2p)
+- [Connection Lifecycle in libp2p](#connection-lifecycle-in-libp2p)
     - [Table of Contents](#table-of-contents)
     - [Overview](#overview)
     - [Definitions](#definitions)
-    - [Protocol Negotiation](#protocol-negotiation)
-        - [multistream-select](#multistream-select)
-    - [Upgrading Connections](#upgrading-connections)
-    - [Opening New Streams Over a Connection](#opening-new-streams-over-a-connection)
+    - [Connection Establishment](#connection-establishment)
+        - [Core Lifecycle](#core-lifecycle)
+        - [Bootstrap Protocol Negociation](#bootstrap-protocol-negociation)
+        - [Advanced Concepts and Manipulation](#advanced-concepts-and-manipulation)
+            - [Extended Protocol Negociation](#extended-protocol-negociation)
+            - [Securing Connections](#securing-connections)
+            - [Multiplexing Streams](#multiplexing-streams)
+        - [Upgrading](#upgrading)
     - [Practical Considerations](#practical-considerations)
-        - [Interoperability](#interoperability)
         - [State Management](#state-management)
             - [Peer Metadata Storage](#peer-metadata-storage)
             - [Connection Limits](#connection-limits)
         - [Connection Lifecycle Events](#connection-lifecycle-events)
     - [Future Work](#future-work)
+    - [Miscellaneous](#miscellaneous)
+        - [A Note about Interoperability](#a-note-about-interoperability)
+        - [Protocol IDs (Extended)](#protocol-ids-extended)
+    
 
 ## Overview
 
-This document describes the process of establishing connections to new peers in
-libp2p and, if necessary, adding security and stream multiplexing capabilities
-to "raw" connections provided by transport protocols.
+This document describes the process and anatomy of establishing connections to new peers in
+libp2p, alongside describing several fundamental aiding protocols and processes. 
 
-We also discuss opening new streams over an existing connection, and the
-protocol negotiation process that occurs to route traffic to the correct
-protocol handler.
+We also discuss the procol negociation processes that help set up sessions, and opening new additional connections over an existing connection.
 
-This document does not cover the establishment of "transport level" connections,
+This document does not cover the establishment of "transport level" connections, 
 for example opening "raw" TCP sockets, as those semantics are specific to each
 transport.
 
 What is covered here is the process that occurs after making the initial
-transport level connection, up to the point where "application level" streams
-are opened, and their protocols are identified and data is routed appropriately
+transport level connection, up to the point where "application level" connections
```

Comment:
Streams is more correct. There are no transport-level connections and application connections. There is one connection, and multiple streams over it.

Reply:
Connections is more correct, "streams" is a concept born from multiplexing, this is concerning itself with the lowest levels of the protocol, where multiplexing is a higher concept which simply provides more connections (streams) over one connection.